friends/src/components/Form.js

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+
+class Form extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: "",
+      age: 0,
+      email: ""
+    };
+  }
+  componentDidMount() {
+    if (this.props.edit) {
+      this.setState({
+        name: this.props.name,
+        age: this.props.age,
+        email: this.props.email
+      });
+    }
+  }
+  changeHandler = e => {
+    this.setState({
+      [e.target.name]: e.target.value
+    });
+  };
+  submitHandler = e => {
+    e.preventDefault();
+    if (this.props.edit) {
+      this.props.editFriend(this.state, this.props.id);
+    } else {
+      this.props.addNewFriend(this.state);
+    }
+  };
+  render() {
+    return (
+      <div>
+        <h1>{this.props.edit ? "Edit Friend" : "Add New friend"}</h1>
+        <form onSubmit={this.submitHandler}>
+          Name:
+          <br />
+          <input
+            onChange={this.changeHandler}
+            type="text"
+            name="name"
+            value={this.state.name}
+          />
+          <br />
+          Age:
+          <br />
+          <input
+            onChange={this.changeHandler}
+            type="number"
+            name="age"
+            value={this.state.age}
+          />
+          <br />
+          Email:
+          <br />
+          <input
+            onChange={this.changeHandler}
+            type="text"
+            name="email"
+            value={this.state.email}
+          />
+          <br />
+          <br />
+          <input
+            type="submit"
+            value={this.props.edit ? "Edit Friend" : "Add New friend"}
```

Comment:
...and also here. Why not make a variable for it before the return in your render?
```javascript
const title = this.props.edit ? 'Edit Friend' : 'Add New friend';

and then use it in your h1 and here:
```javascript
<h1>{title}</h1>
// ...
<input
  type="submit"
  value={title}
/>
```
```